### ISSUE TITLE
'cv' parameter range error for IV algorithms

### ISSUE THREAD

**Issue by canersoy** [py-why/EconML#762]
I am receiving the following error on the 'est.fit(Y=Y, T=T, Z=Z, X=X)' line for all of the econml IV algorithms. I've changed the 'cv' parameter manually, but still received the same error.

Error:
```
Exception has occurred: InvalidParameterError
The 'cv' parameter of LogisticRegressionCV must be an int in the range [2, inf), an object implementing 'split' and 'get_n_splits', an iterable or None. Got <econml.sklearn_extensions.model_selection.WeightedStratifiedKFold object at 0x0000018503EE3C10> instead.
```

Code:
```
np.random.seed(123)
from econml.iv.dml import DMLIV
est = DMLIV(discrete_treatment=True, discrete_instrument=True)
est.fit(Y=Y, T=T, Z=Z, X=X)
```

**Comment by fverac** [comment]
Thanks for sharing this issue. Though I cannot reproduce it locally with the following code:
``` 
import numpy as np
np.random.seed(123)
from econml.iv.dml import DMLIV

Y = np.random.normal(size=(100,1))
T = np.random.binomial(n=1, p=0.5, size=(100,1))
Z = np.random.binomial(n=1, p=0.5, size=(100,1))
X = np.random.normal(size=(100,1))

est = DMLIV(discrete_treatment=True, discrete_instrument=True)
est.fit(Y=Y, T=T, Z=Z, X=X)
```

Could you share the platform you are using and the output of `pip list`?

**Comment by canersoy** [comment]
Hello,

I am using VS Code. My pip list output is as follows:

absl-py                      1.4.0
astunparse                   1.6.3
autopep8                     2.0.2
cachetools                   5.3.0
certifi                      2022.12.7
charset-normalizer           3.1.0
click                        8.1.3
cloudpickle                  2.2.1
colorama                     0.4.6
colormath                    3.0.0
contourpy                    1.0.7
cycler                       0.11.0
econml                       0.14.0
flatbuffers                  23.3.3
fonttools                    4.39.2
gast                         0.4.0
google-auth                  2.17.2
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
grf                          0.2.1
grpcio                       1.53.0
h5py                         3.8.0
idna                         3.4
jax                          0.4.8
joblib                       1.2.0
keras                        2.12.0
kiwisolver                   1.4.4
libclang                     16.0.0
lightgbm                     3.3.5
llvmlite                     0.39.1
Markdown                     3.4.3
MarkupSafe                   2.1.2
matplotlib                   3.7.1
ml-dtypes                    0.0.4
networkx                     3.1
nltk                         3.8.1
nml                          0.6.1
numba                        0.56.4
numpy                        1.22.0
oauthlib                     3.2.2
opt-einsum                   3.3.0
packaging                    23.0
pandas                       1.5.3
patsy                        0.5.3
Pillow                       8.4.0
pip                          22.3.1
ply                          3.11
protobuf                     4.22.1
pyaml                        21.10.1
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycodestyle                  2.10.0
pyparsing                    3.0.9
python-dateutil              2.8.2
pytz                         2022.7.1
PyYAML                       6.0
Werkzeug                     2.2.3
wheel                        0.40.0
wrapt                        1.14.1
xgboost                      1.7.5

**Comment by canersoy** [comment]
Hello again,

I've tried my code on another computer and it worked fine. So, as you suggested, it must be because of package conflicts. I wonder which package is causing a problem for my case?

**Comment by fverac** [comment]
Quickly experimenting with trying to use packages from your `pip list` I am still not able to get an error on my end. Though in setting up my environment I used `conda create -n dmliv_bug python=3.8` which initialized the environment in a way that might have hidden or fixed what is causing your issue.

Simplest answer for you might be to try and create a fresh environment on your original computer to see if it changes things. 

You can also additionally share your sklearn version since it wasn't included in your `pip list`. The version I have is 1.1.3.

**Comment by canersoy** [comment]
Thank you very much.
